types/joint.d.ts

```diff
@@ -454,299 +452,310 @@ export namespace dia {
 
-        localToPageRect(x: number | g.Rect, y?: number, width?: number, height?: number): g.Rect;
+        protected renderMarkup(): void;
+    }
 
-        localToPaperPoint(x: number | g.Point, y?: number): g.Point;
+    class LinkView extends CellViewGeneric<Link> {
+        options: {
+            shortLinkLength?: number,
+            doubleLinkTools?: boolean,
+            longLinkLength?: number,
+            linkToolsOffset?: number,
+            doubleLinkToolsOffset?: number,
+            sampleInterval: number
+        };
 
-        localToPaperRect(x: number | g.Rect, y?: number, width?: number, height?: number): g.Rect;
+        getConnectionLength(): number;
 
-        matrix(): SVGMatrix;
+        sendToken(token: SVGElement, duration?: number, callback?: () => void): void;
 
-        matrix(ctm: SVGMatrix | Vectorizer.Matrix): this;
+        addVertex(vertex: Point): number;
 
-        pageOffset(): g.Point;
+        getPointAtLength(length: number): g.Point; // Marked as public api in source but not in the documents
 
-        pageToLocalPoint(x: number | g.Point, y?: number): g.Point;
+        update(model: Cell, attributes: object, options?: object): this;
 
-        pageToLocalRect(x: number | g.Rect, y?: number, width?: number, height?: number): g.Rect;
+        protected mouseenter(evt: Event): void;
 
-        paperToLocalPoint(x: number | g.Point, y?: number): g.Point;
+        protected mouseleave(evt: Event): void;
 
-        paperToLocalRect(x: number | g.Rect, y?: number, width?: number, height?: number): g.Rect;
+        protected onEndModelChange(endType: 'source' | 'target', endModel?: Element,
+                                   opt?: { cacheOnly?: boolean, handleBy?: string, translateBy?: boolean, tx?: number, ty?: number }): void;
 
-        remove(): this;
+        protected onLabelsChange(): void;
 
-        render(): this;
+        protected onSourceChange(cell: Cell, sourceEnd: { id: string }, options: object): void;
 
-        scale(): Vectorizer.Scale;
-        scale(sx: number, sy?: number, ox?: number, oy?: number): this;
+        protected onTargetChange(cell: Cell, targetEnd: { id: string }, options: object): void;
 
-        scaleContentToFit(options?: ScaleContentOptions): void;
+        protected onToolsChange(): void;
 
-        setDimensions(width: number, height: number): void;
+        // changed is not used in function body.
+        protected onVerticesChange(cell: Cell, changed: any, options: object): void;
 
-        setGridSize(gridSize: number): this;
+        protected pointerdown(evt: Event, x: number, y: number): void;
 
-        setInteractivity(value: any): void;
+        protected pointermove(evt: Event, x: number, y: number): void;
 
-        setOrigin(x: number, y: number): this;
+        protected pointerup(evt: Event, x: number, y: number): void;
+    }
 
-        snapToGrid(x: g.Point | number, y?: number): g.Point;
+    interface Paper extends mvc.View {
```

Comment:
Paper is not an interface, it is a class.  Also if we make mvc.View generic (see below) this should be:

```TypeScript
class Paper extends mvc.View<Graph> {
```